include/boost/uuid/detail/random_provider_bcrypt.ipp

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright (c) 2017 James E. King III
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+//   http://www.boost.org/LICENCE_1_0.txt)
+//
+// BCrypt provider for entropy
+//
+
+#include <boost/core/ignore_unused.hpp>
+#include <boost/winapi/bcrypt.hpp>
+#include <boost/winapi/get_last_error.hpp>
+#include <boost/throw_exception.hpp>
+#include <string>
+
+#if defined(BOOST_UUID_SOURCE) || (!defined(BOOST_ALL_NO_LIB) && !defined(BOOST_UUID_RANDOM_PROVIDER_NO_LIB))
```

Comment:
`BOOST_UUID_SOURCE` is by convention set when the source of the library is compiled - which we don't have as we're header-only. So it's not quite the right macro name to use here.

Reply:
Okay I will try to switch to adding the lib in the jamfile.

Reply:
I renamed the definition to `BOOST_UUID_FORCE_AUTO_LINK` and kept the same logic.  It does not have to be put into the documentation.

Reply:
That's probably easiest, otherwise the selection logic would have to be replicated in the Jamfile which would probably be a challenge. :-)

Reply:
We could link to `advapi32` regardless as everyone links to it anyway, WinCE probably not a concern because I suspect not many use Boost.Build there, but I'm not sure how to decide whether to link to `bcrypt`.

Reply:
These auto-link statements are designed to make it easier for folks consuming Boost.Uuid regardless of the build system.